src/sentry/models/userrole.py

```diff
@@ -0,0 +1,34 @@
+from typing import FrozenSet
+
+from django.db import models
+
+from sentry.db.models import ArrayField, Model, sane_repr
+
+
+class UserRole(Model):
+    """
+    Roles are applied to administrative users and apply a set of `UserPermission`.
+    """
+
+    __include_in_export__ = True
+
+    name = models.CharField(max_length=32, unique=True)
+    permissions = ArrayField()
+    users = models.ManyToManyField("sentry.User")
```

Comment:
anyone aware of this is gonna cause issues?

Reply:
Our typical pattern here is to manually create the bridging table model. I don't think it's required though. The generated SQL looks like it has produced the bridging table automatically just fine.

I guess the only real reason to do it manually would be if you want to have any custom properties for a specific user/permission combo.

Reply:
Does the generated join table have indexes or constraints on the columns?

Reply:
Is it ok for a user to have multiple roles?

Reply:
mmm actually might be valuable to have the `date_added` property on the m2m row. will give some more thought this afternoon. we'll put audit logs in place either way so may not be a big deal.

Reply:
Adding a constraint will take a lock on `auth_users` but it should be a short one. If you wanted to play it safe you could add a `NOT VALID` constraint and validate it separately.

```sql
ALTER TABLE "sentry_userrole_users" ADD CONSTRAINT "sentry_userrole_user_userrole_id_cdeed4a6_fk_sentry_us" FOREIGN KEY ("user_id") REFERENCES "auth_users" ("id") NOT VALID;
ALTER TABLE "sentry_userrole_users" VALIDATE "sentry_userrole_user_userrole_id_cdeed4a6_fk_sentry_us"
```